routes/api/users.js

```diff
@@ -0,0 +1,145 @@
+const express = require("express");
+const bcrypt = require("bcryptjs");
+const Joi = require("joi");
+const jwt = require("jsonwebtoken");
+const User = require("../../service/models/user.js");
+const passport = require("passport");
+const { ExtractJwt, Strategy: JwtStrategy } = require("passport-jwt");
+const dotenv = require("dotenv");
+dotenv.config();
+
+const router = express.Router();
+const secret = process.env.SECRET;
+
+const userSchema = Joi.object({
+  password: Joi.string().min(6).required().messages({
+    "any.required": "Password is required",
+  }),
+  email: Joi.string().email().required().messages({
+    "any.required": "Email is required",
+    "string.email": "Invalid email format",
+  }),
+  subscription: Joi.string()
+    .valid("starter", "pro", "business")
+    .default("starter"),
+  token: Joi.string().allow(null).default(null),
+});
+
+router.post("/users/register", async (req, res) => {
+  try {
+    const { error } = userSchema.validate(req.body);
+    if (error) {
+      return res.status(400).json({ message: "Validation error", error });
+    }
+
+    const { email, password } = req.body;
+
+    const existingUser = await User.findOne({ email });
+    if (existingUser) {
+      return res.status(409).json({ message: "Email in use" });
+    }
+
+    const salt = await bcrypt.genSalt(10);
+    const hashedPassword = await bcrypt.hash(password, salt);
+
+    const newUser = new User({
+      email,
+      password: hashedPassword,
+      subscription: "starter",
+    });
+
+    await newUser.save();
+
+    res
+      .status(201)
+      .json({ user: { email, subscription: newUser.subscription } });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Something went wrong" });
+  }
+});
+
+router.post("/users/login", async (req, res) => {
+  try {
+    const { error } = userSchema.validate(req.body);
+    if (error) {
+      return res.status(400).json({ message: "Validation error", error });
+    }
+
+    const { email, password } = req.body;
+
+    const user = await User.findOne({ email });
+    if (!user) {
+      return res.status(401).json({ message: "Email or password is wrong" });
+    }
+
+    const isPasswordValid = await bcrypt.compare(password, user.password);
+    if (!isPasswordValid) {
+      return res.status(401).json({ message: "Email or password is wrong" });
+    }
+
+    const token = jwt.sign({ user_id: user._id, email }, process.env.SECRET, {
+      expiresIn: "1h",
+    });
+
+    user.token = token;
+    await user.save();
+
+    res.status(200).json({
+      token,
+      user: { email, subscription: user.subscription },
+    });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Something went wrong" });
+  }
+});
+
+passport.use(
```

Comment:
To powinno być uruchomione gdzieś w głównym flow w app.js, nad routingiem. Polecam też umieścić go w innym pliku a potem importować do app.js